### PR TITLE
helm: add certs secret mount

### LINKED ISSUE #4639: helm: support mounting CA certs into coder pod via Helm
we currently do this in v1 via the `certs.secrets` value. the current workaround is to build the coder image with the CA bundle mounted into it. e.g.

```
FROM ghcr.io/coder/coder

COPY ca-bundle.crt /etc/ssl/certs/ca-certificates.crt

RUN apk --update --no-cache add ca-certificates; \
		update-ca-certificates -v

USER coder
```

### PR DESCRIPTION
**Pull request by ericpaulsen**
adds a volume & corresponding mount for CA certificates. a prospective customer needed this today when configuring their air-gapped deploy.

closes #4639

### CODE REVIEW

File: helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -98,6 +98,11 @@ volumes:
   secret:
     secretName: {{ .Values.coder.tls.secretName | quote }}
 {{- end }}
+{{- if .Values.coder.certs.secret.name }}
```
Comment (deansheather) [inline helm/templates/_helpers.tpl:101]: needs to be added to the if statement at the top of the block sadly

File: helm/templates/_helpers.tpl [hunk 2]
```diff
@@ -119,6 +124,12 @@ volumeMounts:
   mountPath: "/etc/ssl/certs/coder/{{ .Values.coder.tls.secretName }}"
   readOnly: true
 {{- end }}
+{{- if .Values.coder.certs.secret.name }}
```
Comment (deansheather) [inline helm/templates/_helpers.tpl:127]: needs to be added to the if statement at the top of the block

File: helm/values.yaml [hunk 3]
```diff
@@ -68,6 +68,14 @@ coder:
     #   cpu: 100m
     #   memory: 128Mi
 
+  # coder.certs -- Certificate to mount inside the Coder pod (e.g. CA bundle).
+  certs:
+    secret:
+      # coder.certs.secret.name -- Name of the secret.
+      name: ""
+      # coder.certs.secret.key -- Key pointing to a certificate in the secret.
+      key: ""
```
Comment (deansheather) [inline helm/values.yaml:77]: we should probably make this accept a list instead of a single secret

**Comment by ericpaulsen** [comment]
@deansheather i updated it to use a list of secrets, using `range` to loop through the list. let me know if I missed anything here.

**Comment by deansheather** [comment]
@ericpaulsen Yes, it will have to be a list of objects with the name and key field.

**Comment by deansheather** [comment]
I think

**Review by deansheather — APPROVED** [review]
once tested LGTM but since I contributed you should get a second opinion :D

**Comment by ericpaulsen** [comment]
@deansheather I tested with `helm template`. Looks good!

**Review by coadler — APPROVED** [review]

**Pull request merged**